Sources/Components/Radio/Radio.swift

```diff
@@ -0,0 +1,136 @@
+import SwiftUI
+
+extension Warp {
+    /// Radio buttons allow users to select one option from a set.
+    ///
+    /// `Radio` is a customizable component that allows users
+    /// to create a list of radio buttons where only one can be selected at any given time.
+    ///
+    /// - Parameters:
+    ///   - selectedOption: A binding to the currently selected option.
+    ///   - options: An array of options that conform to `Identifiable` and `Hashable`.
+    ///   - label: A closure that provides a label for each option.
+    ///   - state: The state of the radio button group (default, error, disabled).
+    public struct Radio<Option: Identifiable & Hashable>: View {
+        /// A binding to the currently selected option.
+        @Binding var selectedOption: Option
+        /// An array of options that conform to `Identifiable` and `Hashable`.
+        var options: [Option]
+        /// A closure that provides a label for each option.
+        var label: (Option) -> String
+        /// The state of the radio button group (default, error, disabled).
+        var state: RadioButtonState
```

Comment:
```suggestion
        var radioButtonState: RadioButtonState
```
🤔 is there a chance that people will confuse `state` for something that is expected to be a `@State` variable?